MdeModulePkg/Core/Dxe/Misc/MemoryProtectionSupport.c

```diff
@@ -3355,21 +3355,21 @@ InitializePageAttributesForMemoryProtectionPolicy (
         LShiftU64 (MemoryMapEntry->NumberOfPages, EFI_PAGE_SHIFT),
         MemoryMapEntry->Attribute
         );
+    }
 
-      // Add EFI_MEMORY_RP attribute for the first page of the stack if stack
-      // guard is enabled.
-      if ((StackBase != 0) &&
-          ((StackBase >= MemoryMapEntry->PhysicalStart) &&
-           (StackBase <  MemoryMapEntry->PhysicalStart +
-            LShiftU64 (MemoryMapEntry->NumberOfPages, EFI_PAGE_SHIFT))) &&
-          gDxeMps.CpuStackGuard)
-      {
-        SetUefiImageMemoryAttributes (
-          StackBase,
-          EFI_PAGES_TO_SIZE (1),
-          EFI_MEMORY_RP | MemoryMapEntry->Attribute
-          );
-      }
+    // Add EFI_MEMORY_RP attribute for the first page of the stack if stack
+    // guard is enabled.
+    if ((StackBase != 0) &&
+        ((StackBase >= MemoryMapEntry->PhysicalStart) &&
+         (StackBase <  MemoryMapEntry->PhysicalStart +
+          LShiftU64 (MemoryMapEntry->NumberOfPages, EFI_PAGE_SHIFT))) &&
+        gDxeMps.CpuStackGuard)

```

Comment:
Very minor, but ideally this would be short circuted to the stack guard feature such that the other conditions are not evaluated if it is not enabled.

```suggestion
    if (gDxeMps.CpuStackGuard &&
        (StackBase != 0) &&
        ((StackBase >= MemoryMapEntry->PhysicalStart) &&
         (StackBase <  MemoryMapEntry->PhysicalStart +
          LShiftU64 (MemoryMapEntry->NumberOfPages, EFI_PAGE_SHIFT))))
```